scripts/webpack.conf.js

```diff
@@ -1,47 +1,76 @@
 const { modifyWebpackConfig, shallowMerge, defaultOptions } = require('@gera2ld/plaid');
+const { isProd } = require('@gera2ld/plaid/util');
 const webpack = require('webpack');
 const WrapperWebpackPlugin = require('wrapper-webpack-plugin');
+const TerserPlugin = require('terser-webpack-plugin');
 const projectConfig = require('./plaid.conf');
 
 const INIT_FUNC_NAME = 'VMInitInjection';
+const definitions = new webpack.DefinePlugin({
+  'process.env.INIT_FUNC_NAME': JSON.stringify(INIT_FUNC_NAME),
+  'process.env.DEBUG': JSON.stringify(process.env.DEBUG || false),
+});
 
 module.exports = Promise.all([
-  modifyWebpackConfig(async (config) => {
-    config.plugins.push(
-      new webpack.DefinePlugin({
-        'process.env.INIT_FUNC_NAME': JSON.stringify(INIT_FUNC_NAME),
-        'process.env.DEBUG': JSON.stringify(process.env.DEBUG || false),
-      }),
-    );
-    return config;
-  }),
-  modifyWebpackConfig(async (config) => {
+  modify(),
+  modify((config) => {
     config.output.libraryTarget = 'commonjs2';
     config.plugins.push(
       new WrapperWebpackPlugin({
         header: `\
-window.${INIT_FUNC_NAME} = function () {
-  var module = { exports: {} };
-`,
+          window.${INIT_FUNC_NAME} = function () {
+            var module = { exports: {} };
+          `,
         footer: `
-  var exports = module.exports;
-  return exports.__esModule ? exports['default'] : exports;
-};0;`,
+            var exports = module.exports;
+            return exports.__esModule ? exports['default'] : exports;
+          };0;`,
       }),
     );
-    return config;
   }, {
+    'injected-web': {
+      entry: './src/injected/web',
+    },
+  }),
+  modify(null, {
+    injected: {
+      entry: './src/injected',
+    },
+  }),
+]);
+
+function modify(init, pages) {
+  const mergedConfig = shallowMerge(defaultOptions, projectConfig);
+  return modifyWebpackConfig((config) => {
+    config.plugins.push(definitions);
+    if (init) init(config);
+    return config;
+  }, pages && {
     projectConfig: {
-      ...shallowMerge(defaultOptions, projectConfig),
+      ...mergedConfig,
       optimization: {
+        ...mergedConfig.optimization,
         runtimeChunk: false,
         splitChunks: false,
+        minimizer: isProd && [
+          new TerserPlugin({
+            cache: true,
+            parallel: true,
+            sourceMap: true,
+            terserOptions: {
+              compress: {
+                ecma: 6,
+                // 'safe' since we don't rely on function prototypes
+                unsafe_arrows: true,
```

Comment:
I reviewed the source code just now, and found `unsafe_arrows` is safe to be applied for all files in `src` except those in `src/public`. So there is no need to split `inject` from others?

Reply:
I'm still a noob with webpack so you decide, you're the expert :-)

Reply:
Turns out splitting of `injected` strips the big unused things like [request](https://github.com/violentmonkey/violentmonkey/blob/master/src/injected/utils/index.js#L6) (3.5kB difference, 14.5 vs 11). Removing `request` there doesn't help.

Reply:
I also think it's okay to separate `injected` because it's not related to anything in `plaid`.

Reply:
I found `chunkFilter` option in TerserPlugin and updated the PR.